### ISSUE TITLE
Regression in control flow analysis

### ISSUE THREAD

**Issue by MasterQ32** [ziglang/zig#6726]
I found a regression of the control flow analysis via the LoLa CI build:

[This code](https://github.com/MasterQ32/LoLa/blob/1b1a9bc8ebde438f6daf45c9729a714e29cfbbdd/src/library/runtime/environment.zig#L311-L315) fails now:

```zig
comptime var index = 0;
inline while (index < function_info.args.len) : (index += 1) {
    const T = function_info.args[index].arg_type.?;
    zig_args[index] = try convertToZigValue(T, args[index]);
}
```
with this message:
```
./src/library/runtime/environment.zig:314:39: error: control flow attempts to use compile-time variable at runtime
                    zig_args[index] = try convertToZigValue(T, args[index]);
                                      ^
./src/library/runtime/environment.zig:311:38: note: compile-time variable assigned here
                comptime var index = 0;
```

while this one doesn't:
```zig
comptime var index = 0;
inline while (index < function_info.args.len) : (index += 1) {
    const T = function_info.args[index].arg_type.?;
    zig_args[index] = convertToZigValue(T, args[index]) catch unreachable;
}
```

I think this is not the correct behaviour as the control flow itself does not depend on the compile time variable.

Zig Version: 0.6.0+b0684bf08

**Comment by daurnimator** [comment]
I also got this when updating autolua. I ended up swapping

```zig
            inline for (args) |*arg, i| {
                arg.* = check(L, i + 1, @TypeOf(arg.*));
            }
```

for

```zig
            comptime var i = 0;
            inline while (i < args.len) : (i += 1) {
                args[i] = check(L, i + 1, @TypeOf(args[i]));
            }
```

**Comment by Vexu** [comment]
Fixed by 4618c41fa6ca70f06c7e65762d2f38d57b00818c
